data-prepper-plugins/kinesis-source/src/test/java/org/opensearch/dataprepper/plugins/kinesis/source/configuration/KinesisSourceConfigTest.java

```diff
@@ -0,0 +1,148 @@
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+import org.opensearch.dataprepper.pipeline.parser.DataPrepperDurationDeserializer;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.kinesis.common.InitialPositionInStream;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.Reader;
+import java.io.StringReader;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class KinesisSourceConfigTest {
+    private static final String PIPELINE_CONFIG_WITH_ACKS_ENABLED = "pipeline_with_acks_enabled.yaml";
+    private static final String PIPELINE_CONFIG_WITH_POLLING_CONFIG_ENABLED = "pipeline_with_polling_config_enabled.yaml";
+    private static final String PIPELINE_CONFIG_CHECKPOINT_ENABLED = "pipeline_with_checkpoint_enabled.yaml";
+    private static final Duration MINIMAL_CHECKPOINT_INTERVAL = Duration.ofMillis(2 * 60 * 1000); // 2 minute
+
+    KinesisSourceConfig kinesisSourceConfig;
+
+    ObjectMapper objectMapper;
+
+    @BeforeEach
+    void setUp(TestInfo testInfo) throws IOException {
+        String fileName = testInfo.getTags().stream().findFirst().orElse("");
+        final File configurationFile = new File(getClass().getClassLoader().getResource(fileName).getFile());
+        objectMapper = new ObjectMapper(new YAMLFactory());
+        SimpleModule simpleModule = new SimpleModule();
+        simpleModule.addDeserializer(Duration.class, new DataPrepperDurationDeserializer());
+        objectMapper.registerModule(new JavaTimeModule());
+        objectMapper.registerModule(simpleModule);
+
+        final Map<String, Object> pipelineConfig = objectMapper.readValue(configurationFile, Map.class);
+        final Map<String, Object> sourceMap = (Map<String, Object>) pipelineConfig.get("source");
+        final Map<String, Object> kinesisConfigMap = (Map<String, Object>) sourceMap.get("kinesis");
+        String json = objectMapper.writeValueAsString(kinesisConfigMap);
+        final Reader reader = new StringReader(json);
+        kinesisSourceConfig = objectMapper.readValue(reader, KinesisSourceConfig.class);
+
+    }
+
+    @Test
+    @Tag(PIPELINE_CONFIG_WITH_ACKS_ENABLED)
```

Comment:
I don't think we need tagging here. Are these tests particularly slow?

Reply:
I did not notice any slowness with these tests.